python/ray/data/tests/test_dataset.py

```diff
@@ -184,6 +192,55 @@ def mapper(x):
         ds.map(mapper)
 
 
+@pytest.mark.parametrize("lazy", [False, True])
+def test_dataset_lineage_serialization(shutdown_only, lazy):
+    ray.init()
+    ds = ray.data.range(10)
+    ds = maybe_lazy(ds, lazy)
+    ds = ds.map(lambda x: x + 1)
+    ds = ds.map(lambda x: x + 1)
+    ds = ds.random_shuffle()
+    epoch = ds._get_epoch()
+    uuid = ds._get_uuid()
+    plan_uuid = ds._plan._dataset_uuid
+    lazy = ds._lazy
+
+    serialized_ds = ds.serialize_lineage()
+    # Confirm that the original Dataset was properly copied before clearing/mutating.
+    in_blocks = ds._plan._in_blocks
+    # Should not raise.
+    in_blocks._check_if_cleared()
+    if lazy and isinstance(in_blocks, LazyBlockList):
```

Comment:
So we have lazy dataset with non-lazy BlockList? If so can you also check the expectation for this case?

Reply:
Ah good catch as well; we don't support lineage-based serialization for non-lazy blocklists, so we should be asserting that `isinstance(in_blocks, LazyBlockList)` instead of it being part of this conditional.